chips/veer_el2/src/io.rs

```diff
@@ -0,0 +1,93 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright (c) 2024 Antmicro <www.antmicro.com>
+
+use core::fmt::Write;
+use core::ptr::write_volatile;
+use kernel::debug::IoWrite;
+use kernel::hil;
+use kernel::ErrorCode;
+
+pub struct Writer {}
+static mut WRITER: Writer = Writer {};
+
+impl Write for Writer {
+    fn write_str(&mut self, s: &str) -> ::core::fmt::Result {
+        self.write(s.as_bytes());
+        Ok(())
+    }
+}
+
+impl IoWrite for Writer {
+    fn write(&mut self, buf: &[u8]) -> usize {
+        for b in buf {
+            // Print to this address for simulation output
+            unsafe {
+                write_volatile(0xd0580000 as *mut u32, (*b) as u32);
+            }
+        }
+        buf.len()
+    }
+}
+
```

Comment:
This should be in the board `io.rs` file instead of here

Reply:
Yes, they should be separate

Reply:
Alright, I've brought the `struct Writer` back to the board's `io.rs`.